addons/mail/static/src/models/thread_viewer/thread_viewer.js

```diff
@@ -67,6 +67,15 @@ function factory(dependencies) {
             });
         }
 
+        /**
+         * @param {mail.message} message
+         */
+        messageSeen(message) {
```

Comment:
Methods should start with verb.
Function seems to be about receiving a message that is visible by component as a feedback, so naming of function should reflect that.

Reply:
`markMessageAsVisible` ? `notifyMessageIsVisible` ? I have no inspiration ..

Reply:
I think like something like `handleVisible/SeenMessage(message)`. And The `seenMessage` field should be renamed to `lastSeenMessage` (name is a bit ambiguous, so small comment to tell this is the message seen by current partner that is the most recent in current view of thread).

Reply:
I prefer `visible` for my part, as there is a `seen` notion on thread and message which could be confused with this notion I think.

Reply:
Ok for "visible". I have a feeling both notions of "visible" and "seen" should be merged, but that's a topic for another time ;)